pytorch/serve
# Fix  issue#1037 with updated weights

## Description

This PR fixes issue #1037. Following the discussions from PR #1038, that we were not able to load the pre-trained weights for image classification examples on different gpu device (i.e., map_locaiton=self.device/cuda+id), this happened because these weights were serialized using older version of serialization code, and torchvision team is upgrading weights in the master/nightly release now, you can read more [on this issue](https://github.com/pytorch/vision/issues/3767).

This PR, is using the updated weights from vision master branch and using vgg16 as vgg11 was not updated yet.


Fixes #(issue)
#1037 
## Type of change

Please delete options that are not relevant.


## steps to test
```
git clone https://github.com/pytorch/serve.git
git checkout issue_1037_weight_update
python ./ts_scripts/install_dependencies.py --environment=dev --cuda=cu110 # or 102
python ts_scripts/install_from_src.py
```

You can test any of the vision models by following the README, here we use vgg16 example.

```
wget https://download.pytorch.org/models/vgg16-397923af.pth
torch-model-archiver --model-name vgg16 --version 1.0 --model-file ./examples/image_classifier/vgg_16/model.py --serialized-file vgg16-397923af.pth --handler ./examples/image_classifier/vgg_16/vgg_handler.py --extra-files ./examples/image_classifier/index_to_name.json
```

```
mkdir model_store

mv vgg16.mar model_store/

```
In a seprate temrinal:

`torchserve --start --model-store model_store`

Then in a seprate terminal:

```
curl -X POST "localhost:8081/models?model_name=vgg16&url=vgg16.mar&batch_size=1&max_batch_delay=5000&initial_workers=4&synchronous=true"
curl http://127.0.0.1:8080/predictions/vgg16 -T ./examples/image_classifier/kitten.jpg
```
In a separate terminal you want to watch the gpu usage and make sure per each worker there is only one process on all the gpus.

`watch -n01 nvidia-smi`

### Logs of the example run
[run_log.txt](https://github.com/pytorch/serve/files/6437520/run_log.txt)

- UT/IT execution results

- Logs
All working examples are tested on P2.8xlarge model both for eager and traced models.
regression test logs are added. 
[regression_logs.txt](https://github.com/pytorch/serve/files/6437361/regression_logs.txt)

<img width="1792" alt="Screen Shot 2021-05-04 at 11 13 11 AM" src="https://user-images.githubusercontent.com/9162336/117060815-407f5e80-acd6-11eb-8fa1-55850221cf5c.png">




## Checklist:
